auth/user.go

```
@@ -40,18 +48,55 @@ type DroneUser struct {
 	Synced int64  `gorm:"column:user_synced"`
 }
 
+type UserOrganization struct {
+	Role string `gorm:"DEFAULT:\"admin\""`
+}
+
+//Organization struct
+type Organization struct {
+	ID        uint                 `gorm:"primary_key" json:"id"`
+	CreatedAt time.Time            `json:"createdAt"`
+	UpdatedAt time.Time            `json:"updatedAt"`
+	DeletedAt *time.Time           `sql:"index" json:"deletedAt,omitempty"`
+	Name      string               `gorm:"unique;not null" json:"name"`
+	Users     []User               `gorm:"many2many:user_organizations" json:"users,omitempty"`
+	Clusters  []model.ClusterModel `gorm:"foreignkey:organization_id" json:"clusters,omitempty"`
+}
+
+func (org *Organization) IDString() string {
+	return strconv.FormatUint(uint64(org.ID), 10)
+}
+
 //TableName sets DroneUser's table name
 func (DroneUser) TableName() string {
 	return "users"
 }
 
-func getCurrentUser(req *http.Request) *User {
+func GetCurrentUser(req *http.Request) *User {
 	if currentUser, ok := Auth.GetCurrentUser(req).(*User); ok {
 		return currentUser
 	}
 	return nil
 }
 
+func GetCurrentOrganization(req *http.Request) *Organization {
+	if organization := req.Context().Value(CurrentOrganization); organization != nil {
+		return organization.(*Organization)
+	}
+	return nil
+}
+
+func GetCurrentUserFromDB(req *http.Request) (*User, error) {
+	if currentUser, ok := Auth.GetCurrentUser(req).(*User); ok {
+		claims := &claims.Claims{UserID: strconv.Itoa(int(currentUser.ID))}
+		context := &auth.Context{Auth: Auth, Claims: claims, Request: req}
+		if user, err := Auth.UserStorer.Get(claims, context); err == nil {
+			return user.(*User), nil
+		}
+	}
+	return nil, nil
```

shouldn't we return the error here?